src/cargo/core/resolver/mod.rs

```diff
@@ -369,21 +403,32 @@ impl<T> RcVecIter<T> {
     }
 }
 
+impl<T> Clone for RcVecIter<T> {
+    fn clone(&self) -> RcVecIter<T> {
+        RcVecIter {
+            vec: self.vec.clone(),
+            rest: self.rest.clone(),
+        }
+    }
+}
+
 impl<T> Iterator for RcVecIter<T> where T: Clone {
     type Item = (usize, T);
+
```

Comment:
Also historically that `cur` has been critical to debugging problems to resolve, so I would want to keep it if we could!